### ISSUE TITLE
AndroidTV java.langillegalArgumentException: Self-causation not permitted.

### ISSUE THREAD

**Issue by nascentt** [mvysny/photocloud-frame-slideshow#75]
Many thanks for releasing the app on AndroidTV!  It works very well on cloud services such as 500px.  And I am now able to sometimes get streams on samba shares.  However am getting Java.lang.RunTimeException: java.langillegalArgumentException: Self-causation not permitted every other time I try to start a slideshow from a samaba share.

**Comment by mvysny** [comment]
Yes, I have just received the crash report to my mail:
```
Caused by java.lang.IllegalArgumentException: throwable == this
       at sk.baka.photoframe.stream.FileSystemCrawlerKt.rethrow(FileSystemCrawler.kt:207)
       at sk.baka.photoframe.stream.FileSystemCrawler2$1$3.invoke(FileSystemCrawler.kt:112)
```

**Comment by mvysny** [comment]
Fixed in 1.11.1
